lib/sqs_message_handler.rb

```diff
@@ -0,0 +1,44 @@
+require File.join(__dir__, '..', 'boot')
+require File.join(__dir__, 'jobs', 'process_resque_message')
+
+class SQSMessageHandler
+  VALID_ACTIONS = ['update', 'transfer']
+
+  #  options message    [Aws::SQS::Types::Message]
+  #  options sqs_client [Class: Aws::SQS::Client]
+  #  options settings   [Hash]
+  def initialize(options = {})
+    @message    = options[:message]
+    @sqs_client = options[:sqs_client]
+    @logger     = Application.logger
+    @settings   = options[:settings]
+    @parsed_message = {}
+  end
+
+  def handle
+    if old_enough?
```

Comment:
Interesting, I'd assumed this logic would be moved into the Resque handler — so everything goes straight into Redis, and then are worked when appropriate as determined by the Resque handler.

Was this for expediency?

Reply:
(Which is totes fine, obvs)

Reply:
((Though if that's why then might be worth adding a comment))

Reply:
expediency? A little. Also...hmmmmmm...to do that:

1.  I might have to send the whole SQS Message **Object** to Redis, right now I only send the body. I felt _hinkey_ about persisting a serialized object. I'm not sure how idiomatic that is, not saying that it isn't. Just not sure if it is.

2.  Without `resque-retry` or `resque-scheduler` it's a little clumsy to tell a resque message "I'll get back to you." Vanilla resque messages are either run and deleted or throw an exception, which then go into a "Failed" pile. FWIW, messages in the failed pile _can_ be retried via irb, but building retry rules is exactly what `resque-retry` is for.

3.  Since the age of the SQS message is a property of the SQS message - I justified that building business logic around the state of the SQS object is more in the sphere of `SQSMessagehandler` than `ResqueMessageHandler`.